src/runtime/objmodel.cpp

```diff
@@ -3293,8 +3301,9 @@ void rearrangeArguments(ParamReceiveSpec paramspec, const ParamNames* param_name
         getArg(kwargs_idx, oarg1, oarg2, oarg3, oargs) = okwargs;
     }
 
-    if ((!param_names || !param_names->takes_param_names) && argspec.num_keywords && !paramspec.takes_kwargs) {
-        raiseExcHelper(TypeError, "%s() doesn't take keyword arguments", func_name);
+    if ((!param_names || !param_names->takes_param_names) && (argspec.num_keywords || argspec.has_kwargs)
```

Comment:
Not sure if this is the cause of the test failures, but I think it's perfectly allowable to pass kwargs to a non-keyword-taking function if the kwargs dict is empty.